### PR TITLE
Improve spec suite structure, part 1

### PR DESCRIPTION
**Pull request by tombruijn**
PR that updates the test suite to be a little bit more organized. These are the first few things I encountered, but I am working on more. Things that will go in another PR, otherwise this PR will get too big.

- [x] Automatically require spec_helper.rb in the .rspec file.  
  Less code :) Automatic includes :)
- [x] Extract and document `VerySpecificError`  
  It was kind of lost now, defined at the end of the spec_helper, with no idea what it was used for.
- [x] Extract DirectoryHelper module so that we do not define global methods

### CODE REVIEW

**Comment by thijsc** [comment]
👍

**Comment by matsimitsu** [comment]
nice! 👍

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
 
-puts "Runnings specs in #{RUBY_VERSION} on #{RUBY_PLATFORM}"
+Dir[File.join(APPSIGNAL_SPEC_DIR, 'support/helpers', '*.rb')].each do |f|
+  require f
+end
+
+$LOAD_PATH.unshift(File.join(APPSIGNAL_SPEC_DIR, 'support/stubs'))
 
-$LOAD_PATH.unshift(File.join(File.dirname(__FILE__), 'support/stubs'))
+puts "Runnings specs in #{RUBY_VERSION} on #{RUBY_PLATFORM}\n\n"
```
Comment (jeffkreeftmeijer) [inline spec/spec_helper.rb]: "Runnings" -> "Running"

Comment (tombruijn) [inline spec/spec_helper.rb]: wasn't me 🙈

**Pull request merged**